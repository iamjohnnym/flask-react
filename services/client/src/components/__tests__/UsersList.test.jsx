import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import UsersList from '../UsersList';

const users = [
    {
        'id': 1,
        'active': true,
        'username': 'iamjohnnym',
        'email': 'johnny@iamjohnnym.com'
    }
];

test('UsersList renders properly', () => {
    const wrapper = shallow(<UsersList users={users}/>);
    const element = wrapper.find('h4');
    expect(element.length).toBe(1);
    expect(element.get(0).props.children).toBe('iamjohnnym');
});

test('UsersList renders a snapshot properly', () => {
    const tree = renderer.create(<UsersList users={users}/>).toJSON();
    expect(tree).toMatchSnapshot();
});
