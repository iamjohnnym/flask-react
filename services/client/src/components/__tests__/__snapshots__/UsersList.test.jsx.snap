// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UsersList renders a snapshot properly 1`] = `
<div>
  <h4
    className="box title is-4"
  >
    iamjohnnym
  </h4>
</div>
`;
